subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/internal/SourceFoldersCreator.java

```diff
@@ -135,6 +141,23 @@ public String apply(SourceFolder sourceFolder) {
         return entries;
     }
 
+    private Map<SourceSet, String> collectSourceSetOutputPaths(List<SourceSet> sourceSets, String defaultOutputPath) {
```

Comment:
I feel this is a bit too complex. The only clash that can happen here is between a sourceSet name and the default output name. So we don't really need a data structure here, just a simple if-statement.

Reply:
I did this because I wanted to cover a degenerate use case where the deduped output name would clash with another output folder name (see the last integration test). I know it is almost impossible to happen, but for the sake of completeness, I wanted to cover it.